File: keanu-research/PocketKeanu/src/main/java/com/research/pocketKeanu/distributions/ContinuousProbabilisticFunction.java
package com.research.pocketKeanu.distributions;

import com.research.pocketKeanu.abstractTypes.DoubleLike;
import com.research.pocketKeanu.randomFactory.ContinuousRandomFactory;

import java.util.List;
import java.util.function.BiFunction;

public interface ContinuousProbabilisticFunction {
    <T extends DoubleLike<T>> List<T> apply(List<T> in, ContinuousRandomFactory<T> rand);
}

File: keanu-research/PocketKeanu/src/main/java/com/research/pocketKeanu/distributions/ContinuousProbabilisticFunction.kt
package com.research.pocketKeanu.distributions

import com.research.pocketKeanu.abstractTypes.DoubleLike
import com.research.pocketKeanu.randomFactory.DoubleLikeRandomFactory

interface ContinuousProbabilisticFunction {
    fun <T : DoubleLike<T>> apply(inputs : List<T>, rand: DoubleLikeRandomFactory<T>): List<T>
}
